### DISCUSSION TITLE
Application builder milti-token or muliti-application?

### DISCUSSION THREAD

**Discussion by jekeam** [python-telegram-bot/python-telegram-bot#3435] [Q&A] [answered]
How can I run many bots (in one main.py), for example, I have 10 bot tokens, I want to use one database and codebase, that is, I want to run multiple instances of `application.run_polling()`, how should I do it if there are solutions like this:
`applications = Application.builder().tokens([<TOKEN_LIST>]).build()`

**Comment by Bibo-Joshi** [comment, marked as answer]
the Application class is designed in such a way that each instance is responsible for exactly one bot (token). you can run multiple applications in the same script though - have a look at [this wiki section](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Frequently-requested-design-patterns#running-ptb-alongside-other-asyncio-frameworks)

**Reply by jekeam** [reply]
The given example seems outdated, could you please give me an up-to-date example for v20.0a4 ?

**Reply by Bibo-Joshi** [reply]
I wrote that section 2 weeks ago. What about it seems outdated to you?
